dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/controllers/WelcomeController.java

```diff
@@ -1,37 +1,165 @@
+import nl.tudelft.dnainator.javafx.widgets.dialogs.ExceptionDialog;
+import nl.tudelft.dnainator.javafx.widgets.dialogs.ProgressDialog;
 import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ObjectProperty;
 import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import javafx.event.ActionEvent;
 import javafx.fxml.FXML;
+import javafx.scene.control.Button;
 import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
+import javafx.stage.DirectoryChooser;
 
 /**
  * Controller for the welcome screen.
  */
 public class WelcomeController {
+	private static final String SELECT_OPTION = "Select new database...";
+	private ObservableList<String> items;
+	private ObjectProperty<Graph> dbProperty;
 	private BooleanProperty done = new SimpleBooleanProperty(false, "done");
-	@SuppressWarnings("unused") @FXML
-	private ListView list;
-	@SuppressWarnings("unused") @FXML
-	private void startButtonAction(ActionEvent e) {
+	private DBLoadService dbload;
+	private DirectoryChooser dirChooser;
+	private ProgressDialog progressDialog;
+	private static final String DEFAULT_DB_PATH = "target/db";
+	@SuppressWarnings("unused") @FXML private Button startButton;
+	@SuppressWarnings("unused") @FXML private Button deleteButton;
+	@SuppressWarnings("unused") @FXML private Button loadButton;
+	@SuppressWarnings("unused") @FXML private ListView<String> list;
+	
+	@SuppressWarnings("unused") @FXML 
+	private void startButtonAction(ActionEvent a) {
 		done.setValue(true);
 	}
+	
+	@SuppressWarnings("unused") @FXML 
+	private void deleteButtonAction(ActionEvent a) {
+		try {
+			FileUtils.deleteRecursively(new File(getDBPath()));
+			items.remove(getDBPath());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+	
+	@SuppressWarnings("unused") @FXML
+	private void loadButtonAction(ActionEvent e) {
+		progressDialog = new ProgressDialog(list.getParent());
+		if (getDBPath() == SELECT_OPTION && selectDirectory() == null) {
+			return;
+		}
+		dbload.restart();
+	}
 
 	@SuppressWarnings("unused") @FXML
-	private void initialize() {
-		ObservableList<String> items = FXCollections.observableArrayList(
-				"nl/tudelft/DNAinator/db/10_strains",
-				"nl/tudelft/DNAinator/db/38_strains",
-				"nl/tudelft/DNAinator/db/100_strains");
+	private void onMouseClicked(MouseEvent e) {
+		if (e.getClickCount() == 2) {
+			if (getDBPath() == SELECT_OPTION) {
+				File dir = selectDirectory();
+				if (dir == null) {
+					return;
+				}
+				dbload.setDatabase(dir.getAbsolutePath());
+				items.add(dir.getAbsolutePath());
+				list.getSelectionModel().select(getDBPath());
+			} else {
+				progressDialog = new ProgressDialog(list.getParent());
+				dbload.restart();
+			}
+		}
+	}
+	
+	@SuppressWarnings("unused") @FXML
+	private void initialize() throws IOException {
```

Comment:
Still has to be caught somewhere, I'll update to catch it in the method where the IO happens.